scapy/sendrecv.py

```diff
@@ -752,20 +762,54 @@ def bridge_and_sniff(if1, if2, prn=None, L2socket=None, *args, **kargs):
             log_runtime.warning("Argument %s cannot be used in "
                                 "bridge_and_sniff() -- ignoring it.", arg)
             del kargs[arg]
-    if L2socket is None:
-        L2socket = conf.L2socket
-    s1 = L2socket(iface=if1)
-    s2 = L2socket(iface=if2)
+    s1 = None
```

Comment:
You could define a function such as, and reuse it twice:
```
def _init_socket(iface, name):
    if isinstance(iface, SuperSocket):
         s =  iface
         ifn = name
     else:
         if L2socket is None:
           L2socket = conf.L2socket
         s = L2socket(iface=iface)
  return s, ifn

s2, if2 = _init_socket(if2, "iface2")
```

Reply:
Nice one! I had to adjust this code because Python 2 lacks the `nonlocal` keyword that would be useful here for `L2socket`, but that's great! (I wanted to factor out this code but never could find something good enough)